ext/standard/var_unserializer.re

```diff
@@ -1193,6 +1193,96 @@ object ":" uiv ":" ["]	{
 	return object_common(UNSERIALIZE_PASSTHRU, elements, has_unserialize);
 }
 
+"E:" uiv ":" ["]	{
+	size_t len, maxlen;
+	char *str;
+	zend_string *full_name, *enum_name, *case_name;
+	zend_class_entry *ce;
+
+    if (!var_hash) return 0;
+
+	len = parse_uiv(start + 2);
+	maxlen = max - YYCURSOR;
+	if (maxlen < len || len == 0) {
+		*p = start + 2;
+		return 0;
+	}
+
+	str = (char*)YYCURSOR;
+
+	YYCURSOR += len;
+
+	if (*(YYCURSOR) != '"') {
+		*p = YYCURSOR;
+		return 0;
+	}
+	if (*(YYCURSOR+1) != ';') {
+		*p = YYCURSOR+1;
+		return 0;
+	}
+
+	full_name = zend_string_init(str, len, 0);
```

Comment:
It's inefficient to allocate a string and then immediately free it - the original won't change. Can just use `str[i]` and strlen instead of `ZSTR_*`

Reply:
This comment still applies

Reply:
We're using `full_name` in an error now, so maybe not.